### DISCUSSION TITLE
Can not set Vertex AI API

### DISCUSSION THREAD

**Discussion by d2clab** [langgenius/dify#7975] [Help]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing? Tell me about your story.

Couldn't set Vertex AI | Google Cloud Platform API.
Error：'utf-8' codec can't decode byte 0xed in position 0: invalid continuation byte
Pretty sure that Project ID* and Service Account Key are right. Because of the location?

**Comment by neyec** [comment]
Hi. I was having the same issue because I was copying the private key inside the JSON file. I managed to configure the Vertex Model by using these settings:

**Project-ID**: your-project-id
**Location**: us-east1
**Service Account Key**: Encode the whole JSON text file to base64 (the file generated in IAM->Service Accounts->Create Service Account->Keys->Add Key->Create New Key-> JSON). You can use Python to do that, or this [website](https://www.base64encode.org/) for instance.

**Comment by d2clab** [comment]
Yep! That's the way. I figured it out two months ago myself. I wonder if
others know this.. this is not in any documentation.

**Comment by neyec** [comment]
Exactly. And if someone is trying to use Application Default Credentials this is how to (0 in base64 is IDA= ) :

<img width="719" alt="image" src="https://github.com/user-attachments/assets/cae4defe-be88-46fd-a25f-aee0c0f16b1f">

**Comment by d2clab** [comment]
Thanks for the tip!
